sse.go

```diff
@@ -128,7 +138,7 @@ func (s *Server) Restart() {
 
 // Shutdown performs a graceful server shutdown.
 func (s *Server) Shutdown() {
-	s.shutdown <- true
+	close(s.shutdown)
```

Comment:
Why does this need to change? If no message is sent to the shutdown channel, how will the listeners know the shutdown has occurred? (e.g. ln.80)

Reply:
Point #1: Actually, we never need the value sent to the `shutdown` channel. Actually, we can change the type of this channel from `chan bool` to `chan struct{}`.
Point #2: When we close the channel, then all the readers from this channel (<-s.shutdown) receive the default value (in this case `false` value).

Conclusion: closing the `shutdown` channel will work as a mechanism of informing all the goroutines about the sse server shutdown process.

Reply:
Ah oh, I am not experienced in using channels and I didn't know the default value got sent when a channel is closed.